uktrade/data-hub-frontend
# Fix investment project required fields

## Description of change

Not all fields for investment projects that are required by the API were marked as required by the front end. This caused a cryptic message `"Could not load TASK_EDIT_INVESTMENT_PROJECT_REQUIREMENTS [Retry] [Dismiss]"` to be displayed to the user when the API required a field for which no value was provided.

This PR adds frontend field-level validation for the three investment project forms:

- Edit summary
- Edit requirements and locations
- Edit value

## Test instructions

When editing any part of an investment project (summary, requirements, or value), depending on the project's current stage and the stage a specific field is required, users should see one of:

1. The text `(optional)` added to fields that are optional at the current and next stage. These fields can be empty when saving the form.
2. The text `(required at next stage)` added to fields that are required at the next stage, and thus need a value before the project can move to the next stage. These fields can be empty when saving the form. These fields should also be listed in the blue info box when viewing a project.
3. The text `(required)` added to fields that are required at the current stage. If these fields don't have values at this stage, when attempting to save the form, these should raise a validation error and have red warning boxes appear. This is instead of the error `"Could not load TASK_EDIT_INVESTMENT_PROJECT_REQUIREMENTS [Retry] [Dismiss]"`.

Stages that individual fields are required at are specified in the `GET_REQUIRED_FIELDS_AFTER_STAGE` constant. These should match the API's `get_required_fields_after_stage` list.

> [!NOTE]
> Part of this work was already reviewed in #7269. However, this PR addresses bugs that were introduced in that PR.

## Screenshots

### Before

![image](https://github.com/user-attachments/assets/59a72a36-aa06-4da5-8915-a7ee59c03adb)

### After

<img width="823" alt="image" src="https://github.com/user-attachments/assets/ddb86526-d424-4fdb-a56b-72ee3a61117d">
<img width="855" alt="image" src="https://github.com/user-attachments/assets/6854ce75-b86d-4810-9ef5-35b1e97cbb15">

## Checklist

[//]: # 'When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/main/docs/Code%20review%20guidelines.md'
